src/Swift.Runtime/tests/LibraryTests/SwiftArrayTests.cs

```diff
@@ -0,0 +1,190 @@
+    public SwiftArrayTests(TestFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    public class TestFixture
+    {
+        static TestFixture()
+        {
+        }
+
+        private static void InitializeResources()
+        {
+        }
+    }
+
+    [Fact]
+    static void SmokeTest()
+    {
+        var metadata = TypeMetadata.GetTypeMetadataOrThrow<SwiftArray<int>>();
+        Assert.Equal((nuint)8, metadata.Size);
+
+        var array = new SwiftArray<int>();
+        Assert.Equal(0, array.Count);
+    }
+
+    [Fact]
+    static void Append()
+    {
+        var array = new SwiftArray<int>();
+        array.Append(42);
+        array.Append(17);
+        var c = array.Count;
+        Assert.Equal(2, c);
+        Assert.Equal(17, array[1]);
+    }
+
+    [Fact]
+    static void Insert()
+    {
+        var array = new SwiftArray<int>();
+        array.Append(42);
+        array.Append(17);
+        var c = array.Count;
+        Assert.Equal(2, c);
+        array.Insert(1, 99);
+        c = array.Count;
+        Assert.Equal(3, c);
+        Assert.Equal(17, array[2]);
+    }
+
+    [Fact]
+    static void Replace()
+    {
+        var array = new SwiftArray<int>();
+        array.Append(42);
+        array.Append(17);
+        var c = array.Count;
+        Assert.Equal(2, c);
+        Assert.Equal(17, array[1]);
+        array[1] = 99;
+        Assert.Equal(99, array[1]);
+    }
+
+    [Fact]
+    static void Remove()
+    {
+        var array = new SwiftArray<int>();
+        array.Append(42);
+        array.Append(17);
+        var c = array.Count;
+        Assert.Equal(2, c);
+        array.Remove(0);
+        c = array.Count;
+        Assert.Equal(1, c);
+        Assert.Equal(17, array[0]);
+    }
+
+    [Fact]
+    static void Clear()
+    {
+        var array = new SwiftArray<int>();
+        array.Append(42);
+        array.Append(17);
+        var c = array.Count;
+        Assert.Equal(2, c);
+        array.RemoveAll();
+        c = array.Count;
+        Assert.Equal(0, c);
+    }
+
+    [Fact]
+    public void OutOfRangeAccess()
+    {
+        var array = new SwiftArray<int>();
+        Assert.Equal(0, array.Count);
+
+        Assert.Throws<IndexOutOfRangeException>(() =>
+        {
+            var _ = array[0];
+        });
+    }
+
+    [Fact]
+    public void LargeArray()
+    {
+        var array = new SwiftArray<int>();
+        const int count = 1000000;
+        for (int i = 0; i < count; i++)
+        {
+            array.Append(i);
+        }
+        Assert.Equal(count, array.Count);
+
+        Assert.Equal(0, array[0]);
+        Assert.Equal(1, array[1]);
+        Assert.Equal(999999, array[count - 1]);
+    }
+
+    [Fact]
+    public unsafe void ArrayDispose()
+    {
+        var array = new SwiftArray<int>();
+        var payload = *(IntPtr*)array.buffer.storage;
+        var count = Arc.RetainCount(payload);
+
+        array.Dispose();
+
+        Assert.Equal(count - 1, Arc.RetainCount(payload));
```

Comment:
What guarantees that `payload` is still alive at this point?

Reply:
Good point. The test is updated to retain the payload before the dispose. Please let me know if there are better ways to test this.